hcl2template/addrs/plugin.go

```diff
@@ -98,67 +110,71 @@ func IsPluginPartNormalized(str string) (bool, error) {
 //
 // The following are valid source string formats:
 //
-//	name
 //	namespace/name
 //	hostname/namespace/name
 func ParsePluginSourceString(str string) (*Plugin, hcl.Diagnostics) {
-	ret := &Plugin{
-		Hostname:  "",
-		Namespace: "",
-	}
 	var diags hcl.Diagnostics
 
-	// split the source string into individual components
-	parts := strings.Split(str, "/")
-	if len(parts) != 3 {
-		diags = diags.Append(&hcl.Diagnostic{
+	if strings.HasPrefix(str, "/") {
+		return nil, diags.Append(&hcl.Diagnostic{
 			Severity: hcl.DiagError,
-			Summary:  "Invalid plugin source string",
-			Detail:   `The "source" attribute must be in the format "hostname/namespace/name"`,
+			Summary:  "Invalid source URL",
+			Detail:   `The source URL must not start with a '/' character. Ex: "github.com/hashicorp/amazon"`,
 		})
-		return nil, diags
 	}
 
-	// check for an invalid empty string in any part
-	for i := range parts {
-		if parts[i] == "" {
-			diags = diags.Append(&hcl.Diagnostic{
-				Severity: hcl.DiagError,
-				Summary:  "Invalid plugin source string",
-				Detail:   `The "source" attribute must be in the format "hostname/namespace/name"`,
-			})
-			return nil, diags
-		}
+	if strings.HasSuffix(str, "/") {
+		return nil, diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid source URL",
+			Detail:   `The source URL must not end with a '/' character. Ex: "github.com/hashicorp/amazon"`,
+		})
 	}
 
-	// check the 'name' portion, which is always the last part
-	givenName := parts[len(parts)-1]
-	name, err := ParsePluginPart(givenName)
+	if !strings.Contains(str, "/") {
+		return nil, diags.Append(&hcl.Diagnostic{
+			Severity: hcl.DiagError,
+			Summary:  "Invalid source URL",
+			Detail:   `A source URL must at least contain a host and a path. Ex: "github.com/hashicorp/amazon"`,
+		})
+	}
+
+	url, err := url.Parse(str)
 	if err != nil {
-		diags = diags.Append(&hcl.Diagnostic{
+		return nil, diags.Append(&hcl.Diagnostic{
 			Severity: hcl.DiagError,
-			Summary:  "Invalid plugin type",
-			Detail:   fmt.Sprintf(`Invalid plugin type %q in source %q: %s"`, givenName, str, err),
+			Summary:  "Invalid source URL",
+			Detail:   fmt.Sprintf(`A source must be a valid URL: %s. Ex: "github.com/hashicorp/amazon"`, err),
```

Comment:
Anyone can argue that a valid URL does contain x,y,z. Lets just tell the user what is expected. 
```suggestion
			Detail:   fmt.Sprintf(`A source must be in the expected format: %s. Ex: "github.com/hashicorp/amazon"`, err),
```

Reply:
I'm not certain the error message is clearer here tbh, we URL parse the source and return the error from the go library, is "must be in the expected format" clearer than "must be a valid URL"?

Reply:
To be clear, I agree with you that "valid URL" is not the only constraint here and just receiving "must be a valid URL" is not too accurate: I'm not sure about the suggested reword, not that it doesn't need one.

Reply:
Alternatively, what if we returned a list of errors rather than stop at the first one? This way a user will have all the errors that they need to act on. Some need prerequisites to work (like schema detection or query/fragment need a parsed URL), but this may be helpful still?